### ISSUE TITLE
Extract changes/replication filters into separate module

### ISSUE THREAD

**Issue by nolanlawson** [pouchdb/pouchdb#6355]
The recent addition of `pouchdb-selector-core` added 1.5KB min+gz to `pouchdb-core`, which isn't a whole lot but coupled with the fact that `eval()` filters sometimes cause problems for folks trying to run CouchDB in Chrome extensions (or possibly in the future, people using a "no eval allowed" [feature policy](https://github.com/WICG/feature-policy)) leads me to believe it may be valuable to extract a `pouchdb-filter` module that adds those capabilities in, since not everyone may be using them.

Basically I would just like `pouchdb-core` to be as lean as possible, and now that we've extracted `pouchdb-debug` (which saves about 3KB) I'd like to go further. 😄

**Comment by nolanlawson** [comment]
Of course this would be a breaking change, so it would have to fit in 7.0.0.
